Canvas tools refactor

MadsLudvig opened the pull request.

MadsLudvig committed:
Move Zoom

MadsLudvig committed:
Move pan functionality to seperate component

MadsLudvig committed:
Implement draw functionality

MadsLudvig committed:
Refactor Zoom component

MadsLudvig committed:
Delete other unused canvases

MadsLudvig committed:
Insert components from the tools directory in places of old components

MadsLudvig committed:
change import

MadsLudvig committed:
Prettier format

MadsLudvig committed:
Add CanvasMousePosition store

MadsLudvig committed:
Add canvasMouseDown store

MadsLudvig committed:
Add mouseevents for changing store

MadsLudvig committed:
formatting

MadsLudvig committed:
Remove Camera from svgstore and remove id from svg interface

MadsLudvig committed:
use renamed canvas instead

MadsLudvig committed:
Remove unused adapter

krestenlaust commented on src/lib/components/Canvas.svelte:
```diff
@@ -1,128 +1,77 @@
 
-	function stopDrag() {
-		isDragging = false;
+<!-- transform={`scale(${})`} -->
+<!-- svelte-ignore a11y-no-noninteractive-element-interactions -->
```
Er denne kommentar fortsat relevant?
```suggestion
```

krestenlaust commented on src/lib/components/tools/Zoom.svelte:
```diff
@@ -1,33 +1,18 @@
-			}
-			return current;
-		});
+		if ($canvasView.scale >= 200) return;
```
Magic value

krestenlaust commented on src/lib/components/tools/Zoom.svelte:
```diff
@@ -1,33 +1,18 @@
-			return current;
-		});
+		if ($canvasView.scale >= 200) return;
+		$canvasView = { ...$canvasView, scale: $canvasView.scale + 25 };
```
Magic value

krestenlaust commented on src/lib/stores/stateStore.ts:
```diff
@@ -9,37 +9,57 @@ export const viewChat = writable(false);
-	pencil: boolean;
-	erase: boolean;
-	pan: boolean;
-	move: boolean;
+  draw: boolean;
+  erase: boolean;
+  pan: boolean;
+  move: boolean;
 }
```
Kan flere tilstande være slået til på samme tid? Jeg tror måske det er fordelagtigt at bruge en optalt type fremfor et interface og derved få stærkere typer

krestenlaust commented on src/lib/components/tools/Draw.svelte:
```diff
@@ -0,0 +1,69 @@
+
+	function startDraw(mouseDown: boolean) {
+		if (!mouseDown) return;
+		currentId = Math.floor(Math.random() * 1000000);
```
Antager at du senere vil spørge serveren om et ID i stedet

MadsLudvig replied:
Det er faktisk en fejl. Der burde ikke være id

MadsLudvig committed:
Update src/lib/components/Canvas.svelte

Co-authored-by: Kresten Laust <krestenlaust@gmail.com>

MadsLudvig committed:
Convert toolState to enum type

MadsLudvig committed:
Create constants

MadsLudvig committed:
Use index of last element instead of ･ﾟMagic ･ﾟnumber

This is under the assumption that no elements can be removed from the stateStore

MadsLudvig committed:
correct bad enum

MadsLudvig committed:
add needed comment

Many5900 commented on src/lib/components/Menu.svelte:
```diff
@@ -1,23 +1,18 @@
 <script lang="ts">
-	import { toolState, activateTool } from '$lib/stores/stateStore';
+	import { toolState, ToolState } from '$lib/stores/stateStore';
+	// import type { ToolState } from '$lib/stores/stateStore';
```
Why do we have dis?

MadsLudvig committed:
remove comment

Many5900 reviewed (approved):
LGTM!'

krestenlaust commented on src/lib/components/tools/Zoom.svelte:
```diff
@@ -1,33 +1,21 @@
-			return current;
-		});
+		if ($canvasView.scale <= MIN_ZOOM) return;
+		$canvasView = { ...$canvasView, scale: $canvasView.scale - 25 };
```
-25 ?

krestenlaust commented on src/lib/stores/stateStore.ts:
```diff
@@ -9,37 +9,41 @@ export const viewChat = writable(false);
+  y: number;
+  width: number; // Default values, will be replaced on mount
+  height: number; // Default values, will be replaced on mount
+  scale: number;
```
`zoom` or `zoomValue` might have been a more explanatory name

krestenlaust reviewed (commented):
Ellers LGTM
